docs/rst/tutorials/cloud/wan_edge_cloud_tls/wan_edge_cloud_tls.rst

```diff
@@ -0,0 +1,390 @@
+further showing how to secure the edge-cloud communication link with TLS protocol.
+It is recommended to follow these tutorials in order, as some concepts or installations may be already covered.
+
+.. _warning_lan_tls:
+
+.. warning::
+
+    This tutorial is intended for WAN communication.
+    However, if communication through a LAN is your only option, it is still possible to follow the tutorial by changing the ROS 2 Domain Ids so that each ROS 2 node uses a different Domain (``0`` and ``1``).
+    This way the ROS 2 nodes are logically isolated and will not discover other nodes out of their ROS 2 Domain.
+
+Following, all the elements involved in this architecture will be studied, starting with the edge robot, continuing with the controller hosted in the cloud also built as a ROS 2 node and concluding with the intermediate elements that enable communication over the Internet.
+
+The image below describes the scenario presented in this tutorial.
+
+.. figure:: /rst/figures/tutorials/cloud/edge_cloud_wan.png
+   :align: center
+
+Several key elements can be observed in it:
+
+#.  **ROS 2 Application**.
+    *Turtlesim* is the application used for this tutorial.
+    *Turtlesim* is a ROS 2 application, first developed for ROS, aimed at teaching the basic concepts of ROS 2 such as publish/subscribe, services and actions.
+    The edge robot will then be a ``turtlesim_node``, which is a simulator of a robot making use of these communication methods.
+
+#.  **ROS 2 Device Controller**.
+    This is a ROS 2 application that sends commands to the edge robot.
+    A basic C++ application has been developed for this tutorial that sends publications under the topic on which the ``turtlesim_node`` listens.
+    By means of these publications (commands) from the controller, and the feedback information that the controller receives from the ``turtlesim_node``, it is possible to control this node automatically without the need for user intervention which facilitates the deployment of the scenario at hand.
+    The key feature of the *DDS Router* is that it is easy to configure, allowing to connect different networks with ROS 2 applications without requiring to apply any changes to the developer's software or applications.
+
+#.  **ROS 2 Router / DDS Router**.
+    *eProsima ROS 2 Router*, a.k.a `DDS Router <https://github.com/eProsima/DDS-Router>`_, is an end-user software application that enables the connection of distributed ROS 2 networks (see DDS Router documentation `here <https://eprosima-dds-router.readthedocs.io/en/latest/>`_).
+    That is, ROS 2 nodes such as publishers and subscribers, or clients and services, deployed in one geographic location and using a dedicated local network will be able to communicate with other ROS 2 nodes deployed in different geographic areas on their own dedicated local networks as if they were all on the same network through the use of *DDS Router*.
```

Comment:
Comment that this communication is done via internet with TCP, and TLS gives a extra secure layer in this wan communication